Lib/test/test_import/__init__.py

```diff
@@ -111,6 +111,27 @@ def test_from_import_missing_attr_path_is_canonical(self):
         self.assertIn(cm.exception.name, {'posixpath', 'ntpath'})
         self.assertIsNotNone(cm.exception)
 
+    def test_from_import_star_invalid_type(self):
+        TESTFNnoat = TESTFN[1:]
+        source = TESTFNnoat + os.extsep + "py"
+        sys.path.insert(0, os.curdir)
```

Comment:
Perhaps there is a special purposed helper in `test.support` or `test.test_importlib.util` for temporary modifying sys.path and safely clean up.